python/pyspark/sql/pandas/group_ops.py

```diff
@@ -505,22 +514,75 @@ def transformWithStateUDF(
 
             return result
 
-        if isinstance(outputStructType, str):
-            outputStructType = cast(StructType, _parse_datatype_string(outputStructType))
+        def transformWithStateWithInitStateUDF(
+                statefulProcessorApiClient: StatefulProcessorApiClient,
+                key: Any,
+                inputRows: Iterator["PandasDataFrameLike"],
+                # for non first batch, initialStates will be None
+                initialStates: Iterator["PandasDataFrameLike"] = None
+        ) -> Iterator["PandasDataFrameLike"]:
```

Comment:
Can we add some commentss on the possible input combinations that we need to handle in this udf for people to understand easier? IIUC there should be 3 cases:
- Both `inputRows` and `initialStates` contain data. This would only happen in the first batch and the associated grouping key contains both input data and initial state.
- Only `inputRows` contains data. This could happen when either the grouping key doesn't have any initial state to process or it's non first batch.
- Only `initialStates ` contains data. This could happen when the grouping key doesn't have any associated input data but it has initial state to process.

Reply:
Add the above in the comment.